contrib/inventory/vmware_inventory.py

```diff
@@ -614,7 +614,14 @@ def facts_from_proplist(self, vm):
                         lastref = lastref[x]
                     else:
                         lastref[x] = val
-
+        if self.args.debug:
+            self.debugl("For %s" % vm.name)
+            for key in list(rdata.keys()):
+                if isinstance(rdata[key], dict):
```

Comment:
Should this be a recursive method so that it can handle a dict of a dict of a dict (etc) as well?